core/server/ntske.go

```diff
@@ -0,0 +1,97 @@
+package server
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"strconv"
+
+	"go.uber.org/zap"
+
+	"example.com/scion-time/net/ntske"
+)
+
+const defaultNtskePort int = 4460
+
+func runNTSKEServer(log *zap.Logger, listener net.Listener, localHost *net.UDPAddr, provider *ntske.Provider) {
+	for {
+		ke, err := ntske.NewListener(listener)
+		if err != nil {
+			log.Info("failed to accept client", zap.Error(err))
+			break
+		}
+
```

Comment:
I would suggest to factor out the remaining code in the loop into a separate function `handleKeyExchange` and call `defer ke.Conn.Close()` at the beginning of `handleKeyExchange`.

Reply:
Done